producer/src/main/kotlin/streams/StreamsEventRouter.kt

```diff
@@ -0,0 +1,63 @@
+package streams
+
+import org.apache.kafka.clients.producer.Callback
+import org.apache.kafka.clients.producer.Producer
+import org.neo4j.logging.Log
+import streams.events.EntityType
+import streams.events.StreamsEvent
+import org.apache.kafka.common.KafkaException
+import org.apache.kafka.common.errors.AuthorizationException
+import org.apache.kafka.common.errors.OutOfOrderSequenceException
+import org.apache.kafka.common.errors.ProducerFencedException
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.apache.kafka.clients.producer.RecordMetadata
+import streams.serialization.JacksonUtil
+import java.lang.Exception
+
+
+interface StreamsEventRouter {
+
+    fun sendEvent(event : StreamsEvent)
+
+}
+
+
+class StreamsEventRouterImpl(private val log: Log, private val producer: Producer<Long, ByteArray>, private val config: StreamsConfiguration): StreamsEventRouter {
+
+    init {
+        producer.initTransactions()
+    }
+
+    private val producerCallback = { meta: RecordMetadata?, error: Exception? -> log.warn("sending record in partition ${meta?.partition()} offset ${meta?.offset()} data ${meta?.topic()} key size ${meta?.serializedKeySize()}", error) }
+
+    override fun sendEvent(event: StreamsEvent) {
+        val events = when (event.payload.type) {
+            EntityType.node -> NodeRoutingConfiguration.prepareEvent(event, config.nodeRouting)
+            EntityType.relationship -> RelationshipRoutingConfiguration.prepareEvent(event, config.relRouting)
+        }
+        try {
+            log.debug("Trying to send the event with txId ${event.meta.txId} to kafka")
+            producer.beginTransaction()
+            events.forEach {
```

Comment:
we should test this with larger amounts of data, e.g. creating 1M nodes and relationships in one tx

UNWIND range(1,100000) as id create (n:Node {id:id, name:"name "+id})-[:REL {since:id}]->(n);
Needs about 3.5G heap for the transaction.